pkg/client/client.go

```diff
@@ -56,6 +76,27 @@ func New(config *rest.Config, options Options) (Client, error) {
 	if config == nil {
 		return nil, fmt.Errorf("must provide non-nil rest.Config to client.New")
 	}
+	var logger logr.Logger
+
+	if !options.Opts.SuppressWarnings {
+		// surface warnings
+		logger = log.Log.WithName("KubeAPIWarningLogger")
+	} else {
+		// suppress warnings
+		logger = log.NullLogger{}
+	}
+
+	// Set a WarningHandler, the default WarningHandler is log.WarningLogger with
+	// deduplication enabled. See log.WarningLoggerOptions for considerations regarding
+	// deduplication.
+	rest.SetDefaultWarningHandler(
+		log.NewWarningLogger(
+			logger,
+			log.WarningLoggerOptions{
+				Deduplicate: !options.Opts.AllowDuplicateLogs,
+			},
+		),
+	)
```

Comment:
I am wondering a bit if the client is the best place to plug this in. Maybe pkg/cluster would be better? Or both?
Many ppl will use the cache-backed client and all its read operations will not have this enabled since they go through the cache.
cc @vincepri